fvm/src/gas/price_list.rs

```diff
@@ -416,9 +537,91 @@ impl PriceList {
     pub fn on_verify_consensus_fault(&self) -> GasCharge<'static> {
         GasCharge::new("OnVerifyConsensusFault", self.verify_consensus_fault, 0)
     }
+
+    /// Returns the gas required for the specified exec_units.
+    #[inline]
+    pub fn on_consume_exec_units(
+        &self,
+        exec_units: u64,
+    ) -> Result<GasCharge<'static>, anyhow::Error> {
+        Ok(GasCharge::new(
+            "OnConsumeExecUnits",
+            self.gas_per_exec_unit
+                .checked_mul(exec_units as i64)
+                .ok_or_else(|| anyhow!("overflow when consuming exec_units"))?,
+            0,
+        ))
+    }
+
+    /// Converts the specified gas into equivalent exec_units
+    #[inline]
+    pub fn gas_to_exec_units(&self, gas: i64) -> u64 {
+        match self.gas_per_exec_unit {
+            0 => 0,
+            v => (gas / v) as u64,
```

Comment:
So, rounding here is tricky. Especially if we want to be accurate:

1. Round down when setting the fuel limit.
2. When charging gas for fuel:
    1. Convert fuel to gas, rounding up. Charge for that.
    2. Set the fuel snapshot to `gas_to_exec_units(gas_CHARGED)`. This will be larger than the actual fuel used, but will prevent us from double-charging for fuel.

Now, maybe that's overkill. Alternatively, we could just "round up" in case 2 and leave it at that. If the actor uses 1 fuel, makes a syscall, then uses 1 fuel, we'd charge 2 gas (instead of 1 gas). But maybe that doesn't matter?

Reply:
Still TODO.